src/Verify.Tests/VirtualizedRunHelperTests.cs

```diff
@@ -0,0 +1,32 @@
+﻿public class VirtualizedRunHelperTests
```

Comment:
Please add a unit test for testrun on docker as well. - it mounts to `/mnt/approot/<path>`, where `<path>` is your path relative to the solution.
E.g. with the current structure:
```
C:\my-src\contoso\proj-x
                   |-- solution.sln
                   |-- src\...
                   |-- build-outputs
                          |--my-test-binary.dll
```

When running tests from the `my-test-binary.dll` the `Environment.CurrentDirectory` would return `/mnt/approot/build-outputs`

Current code seems to handle this just fine - but it's good to be future proof (e.g. if someone would try to find drive root and map paths from there)

Reply:
I'd as well recommend adding a test for case where `originalCodeBaseRoot` and `currentDir` are more distant then just single direcotry appart.
This would explicitly test a case of more nested project structure (e.g. running tests from proj `C:\my-src\contoso\proj-x\src\data-layer\utils\proj-b.fsproj` while the solution directory is `C:\my-src\contoso\proj-x`)

Reply:
I added the test (but disabled) as I'm not sure how it's supposed to work as there no common part between `/mnt/approot/build-outputs` and `C:\my-src\contoso\proj-x`

Reply:
I might be wrong but I feel like we should on a given assembly test first "solution path" and then fallback on "project path".

Reply:
Good point! - this case (docker run for a build where build outputs are configured to go to the sibling folder of a .sln, while both (.sln and output folder) being directly in the 'application root') was not handled in my original code (it would refuse to map the path and continue with it as is).

While it feels outside of the scope of your original intent - would you want to fix that case as well?
Basically:
 - (already done by the code) First we try to find mapping by finding a 'cross section' of the buildtime and runtime path (e.g. `C:\build\project_dir`, `/mnt/c/build/project_dir/subdir`, have the `build/project_dir` cross section)
 - (missing handling) - we are not able to find the cross section, so we defer the initialization until the first call to `GetMappedBuildPath` requesting to map the path (`pathToMap` in following text). Then 
    - Get the relative build-time path of the `pathToMap` (removing the solution-dir prefix from `pathToMap`)
    - Iterating through current runtime path and trying to append the relative part of `pathToMap` and checking if path exists. Once it exists - we are initialized. If we iterated to the root, we are not able to perform maping for further requests.
    - Sample: current runtime dir: `/mnt/approot/build-outputs`, solution root: `C:\my-src\contoso\proj-x`, `pathToMap`: `C:\my-src\contoso\proj-x\src\data-layer\utils\proj-b.fsproj`.
      - relative requested path:  `src\data-layer\utils\proj-b.fsproj`
      - first attempted path check: `/mnt/approot/build-outputs/src/data-layer/utils/proj-b.fsproj` - doesn't exist
      - second attempted path check: `/mnt/approot/src/data-layer/utils/proj-b.fsproj` - exists, so we have the build time root `C:\my-src\contoso\proj-x\` and runtime map root `/mnt/approot/`


> I might be wrong but I feel like we should on a given assembly test first "solution path" and then fallback on "project path".

Alternatively we can try to use the "project path" (`AttributeReader.TryGetProjectDirectory`) in lieu of the `pathToMap` (from first call to `GetMappedBuildPath`) - and so being able to resolve the second case fully in the constructor, without waiting for the first call to `GetMappedBuildPath`. I guess that's what you meant?
If yes - than it sounds great - I'd just probably leave the door open for fallback from the first call to `GetMappedBuildPath` in case the `AttributeReader.TryGetProjectDirectory` would not return anyhting, or that it would return same path as `AttributeReader.TryGetSolutionDirectory`



Not sure if it sounds reasonable or if you had differnet solution in mind?

Reply:
Yes "project dir" could provide some info to get the common part, but it won't be enough in the case "build output next to .sln".

I'm not sure I will find more time to work on this, is it ok if we merge this PR first?